packages/protocol/contracts/L1/ProofVerifier.sol

```diff
@@ -27,16 +28,21 @@ interface IProofVerifier {
     ) external pure returns (bool verified);
 }
 
-contract ProofVerifier is IProofVerifier {
+contract ProofVerifier is IProofVerifier, EssentialContract {
+    function init(address addressManager) external initializer {
+        EssentialContract._init(addressManager);
+    }
+
     function verifyZKP(
         bytes memory verificationKey,
         bytes calldata zkproof,
         bytes32 blockHash,
         address prover,
         bytes32 txListHash
-    ) external pure returns (bool) {
+    ) external view returns (bool) {
         return
             LibZKP.verify({
+                plonkVerifier: resolve("plonk_verifier", false),
```

Comment:
@Brechtpd If "we can use different circuits setup", does it mean we can remove `verificationKey` from this function and use different `plonkVerifiers`

Reply:
I think it does make sense to use different contracts instead of trying to use verification keys because that may not always work. So we may as well go with the more generalized way that will always work no matter what changes are done to the top circuit.